### PR TITLE
Trivial readme refactor

### PR DESCRIPTION
**Pull request by rffrancon**
* Attempted to clarify language.
* Tidier links.
* Added markdown extension.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+
```
Comment (TheBlueMatt) [inline README.md]: Why the extra lines?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
+Mining Server Proxy
+=====================================
+
+[![Build Status](https://travis-ci.org/TheBlueMatt/mining-proxy.svg?branch=master)](https://travis-ci.org/TheBlueMatt/mining-proxy)
```
Comment (TheBlueMatt) [inline README.md]: Ugh, I hate these things, no external non-plaintext-readable crap, please.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+
+A simple proxy which supports serving work via the Stratum and the [BetterHash](https://github.com/TheBlueMatt/bips/blob/master/bip-XXXX.mediawiki) mining protocols.
+
+Work can be requested from bitcoind using the [2018-02-miningserver](https://github.com/TheBlueMatt/bitcoin/commits/2018-02-miningserver) patchset. Payout information is also available in a similar manner.
```
Comment (TheBlueMatt) [inline README.md]: You lost information (about the protocl used). Also note that pool protocol isnt to bitcoind, its to the pool daemon in this repo.

File: README.md (hunk 1 above)
Comment (rffrancon) [reply README.md]: fixed in latest commits

File: README.md (hunk 2 above)
Comment (rffrancon) [reply README.md]: fixed in latest commits

**Pull request merged**